components/organisms/ChainChecker.tsx

```diff
@@ -0,0 +1,20 @@
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+import { useAccount } from "wagmi";
+
+const ChainChecker = () => {
```

Comment:
As far as I know the libraries we are using, as soon as a user connects a chain that is not supported by the dApp, RainbowKit triggers a modal for chain switching. Have you tested this behavior? I assume it might work and, thus, this file could be removed.

Reply:
@FrancoAguzzi When multiple chains are allowed, the chain-switching modal doesn't appear. So, I followed the approach from the ENS dApp.

Reply:
It would also be helpful to display a 'Not Supported' warning, informing the user so they can make their own decision